### PR TITLE
Azure SC v2

### PR DESCRIPTION
**Pull request by liorblob**
## Status
In Progress

## Related Issues
https://github.com/demisto/etc/issues/19201

## Description
Add v2 to support new auth flow, deprecate the old one

## Does it break backward compatibility?
   - Yes
       - New auth flow

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

## Technical writer review
Mention and link to the files that require a technical writer review.
- [ ] [YAML file](https://github.com/demisto/content/compare/azure-sc?expand=1#diff-1201a468f70b4eea3e58d666e063bb96) - ONLY subscription related params/args
- [ ] [CHANGELOG](https://github.com/demisto/content/compare/azure-sc?expand=1#diff-755c5c818af31699d10a191ecb55478f)

### CODE REVIEW

File: Integrations/AzureSecurityCenter_v2/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1 @@
+New integration - Allow use of different subscriptions, new auth flow. See the integration documentation for more details.
```
Comment (Itay4) [inline Integrations/AzureSecurityCenter_v2/CHANGELOG.md]: no need for changelog

File: Integrations/AzureSecurityCenter_v2/AzureSecurityCenter.yml [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+      type: String
+    - contextPath: Azure.Subscription.Enabled
+      description: Subscription state
+      type: String
+  - deprecated: false
+    description: The location of the responsible ASC of the specific subscription.
+      For each subscription there is only one responsible location.
+    execution: false
+    name: azure-sc-list-location
+  dockerimage: demisto/crypto:1.0.0.303
```
Comment (Itay4) [inline Integrations/AzureSecurityCenter_v2/AzureSecurityCenter.yml]:
```suggestion
  dockerimage: demisto/crypto:1.0.0.3008
```

File: Integrations/AzureSecurityCenter_v2/AzureSecurityCenter.yml [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+  type: 4
+- display: Trust any certificate (not secure)
+  name: unsecure
+  required: false
+  type: 8
+- display: Use system proxy settings
+  name: proxy
+  required: false
+  type: 8
+- display: Default subscription ID to use
```
Comment (Itay4) [inline Integrations/AzureSecurityCenter_v2/AzureSecurityCenter.yml]: place this param above the checkboxes

File: TestPlaybooks/playbook-AzureSCTestPlaybook.yml [hunk 4]
```diff
@@ -0,0 +1,283 @@
+id: Azure SecurityCenter Test
```
Comment (Itay4) [inline TestPlaybooks/playbook-AzureSCTestPlaybook.yml]:
```suggestion
id: Azure SecurityCenter - Test
```

File: TestPlaybooks/playbook-AzureSCTestPlaybook.yml [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+version: -1
+name: Azure SecurityCenter Test
```
Comment (Itay4) [inline TestPlaybooks/playbook-AzureSCTestPlaybook.yml]:
```suggestion
name: Azure SecurityCenter - Test
```

File: Tests/conf.json [hunk 6]
```diff
@@ -2,6 +2,10 @@
     "testTimeout": 160,
     "testInterval": 20,
     "tests": [
+        {
+            "integrations": "Azure Security Center v2",
```
Comment (Itay4) [inline Tests/conf.json]: add another instance - there should be one for and one for prod

**Comment by liorblob** [comment]
@kirbles19 any progress on this?

**Comment by Itay4** [comment]
@richardbluestone please review

**Comment by richardbluestone** [comment]
Reviewed and updated

**Pull request merged**